cvmfs/catalog_traversal.h

```diff
@@ -299,15 +299,15 @@ class CatalogTraversalBase
     const std::string   path;
     const shash::Any    hash;
     const unsigned      tree_level;
-    const unsigned      history_depth;
+    const uint64_t      history_depth;
           CatalogTN    *parent;
 
     // dynamic processing state (used internally)
     std::string   catalog_file_path;
     size_t        catalog_file_size;
     bool          ignore;
     CatalogTN    *catalog;
-    unsigned int  referenced_catalogs;
+    uint64_t      referenced_catalogs;
```

Comment:
if you make this `uint64_t` should then also `tree_level` be `uint64_t`?

Reply:
That should be independent, `tree_level` specifies the "height" of the tree in terms of nested catalogs. The revision represents the timeline of root catalogs.